modules/core/keeper/msg_server.go

```diff
@@ -81,6 +80,15 @@ func (k *Keeper) UpdateClient(goCtx context.Context, msg *clienttypes.MsgUpdateC
 		return nil, err
 	}
 
+	// only check v2 params if this chain is setup with v2 clientKeepr
+	if k.ClientV2Keeper != nil {
+		// check if this relayer is allowed to update if v2 params are set
+		params := k.ClientV2Keeper.GetParams(ctx, msg.ClientId)
+		if !params.IsAllowedRelayer(sdk.MustAccAddressFromBech32(msg.Signer)) {
```

Comment:
Note if params is unset this will pass

Reply:
Should something like this be in membership as well?